Xamarin.PropertyEditing.Mac/Controls/Custom/NumericSpinEditor.cs

```diff
@@ -227,19 +244,25 @@ protected void OnEditingEnded (object sender, EventArgs e)
 			if (!editing) {
 				editing = true;
 				SetValue (numericEditor.StringValue);
-				if (EditingEnded != null)
-					EditingEnded (this, EventArgs.Empty);
-				if (ValueChanged != null)
-					ValueChanged (this, EventArgs.Empty);
+				EditingEnded?.Invoke (this, EventArgs.Empty);
+				ValueChanged?.Invoke (this, EventArgs.Empty);
 				editing = false;
 			}
 		}
 
 		void SetValue (string value)
 		{
-			//Regulates maximun and minium out of range
-			stepper.DoubleValue = CoerceValue (FieldValidation.FixInitialValue (value, Value.ToEditorString ()).ToEditorDouble ());
-			numericEditor.StringValue = FieldValidation.RoundDoubleValue (stepper.DoubleValue.ToEditorString (), NumericMode == ValidationType.Decimal ? FieldValidation.DefaultXcodeMaxRoundDigits : 0);
+			if (AllowRatios) {
+				// We should have a valid ration at this point.
+				var parts = FieldValidation.GetRatioParts (value);
+				SetCommonRatio (FieldValidation.ParseDoubleValue(parts[0]), FieldValidation.ParseDoubleValue (parts[1]), FieldValidation.GetSeparator(value));
```

Comment:
As much of this behavior as possible was supposed to move to the view model, so much so that the `FieldValidation` class could be deleted entirely. This crashes when you just select all and enter a whole value.

Reply:
Oh damn. I will look into this!